src/Storage/Repository/InstanceRepository.cs

```diff
@@ -564,13 +565,17 @@ private static void PreProcess(Instance instance)
         /// - Sets correct LastChanged/LastChangedBy by comparing instance and data elements
         /// </remarks>
         /// <param name="instance">the instance to preprocess</param>
-        private async Task PostProcess(Instance instance)
+        private async Task PostProcess(Instance instance, bool includeDataelements = true)
         {
             Guid instanceGuid = Guid.Parse(instance.Id);
             string instanceId = $"{instance.InstanceOwner.PartyId}/{instance.Id}";
 
             instance.Id = instanceId;
-            instance.Data = await _dataRepository.ReadAll(instanceGuid);
+            if (includeDataelements)
+            {
+                instance.Data = await _dataRepository.ReadAll(instanceGuid);
+            }
+
             if (instance.Data != null && instance.Data.Any())
```

Comment:
Her er det vi tok opp i daily. Dette med at readStatus og lastChanged defineres basert på å sjekke instance og dataElements.

Men slik koden ser ut nå så vil jeg foreslå at sjekken på linje 579 flyttes inn i `includeDataElements`-delen. For den vil alltid evalueres til false hvis man ikke henter ut dataelementer.

Reply:
Dette skjønte jeg ikke. Hvis includeDataelementes==true, så blir jo alt riktig. Og hvis den er false, vil det jo uansett være umulig å få satt read og lastchanged riktig i forhold til å reflektere dataelementene.

Reply:
This would of course require that dataElements be stored in the same container and use the same partition key. Would look something like this:
```
        public async Task<DataElement> Update(DataElement dataElement)
        {
            PartitionKey partitionKey = new PartitionKey(dataElement.InstanceGuid);

            TransactionalBatch batch = Container.CreateTransactionalBatch(partitionKey);

            batch.UpsertItem(dataElement);

            batch.PatchItem("instance_" + dataElement.InstanceGuid, new[]
            {
                PatchOperation.Set("/lastChanged", dataElement.LastChanged),
                PatchOperation.Set("/lastChangedBy", dataElement.LastChangedBy),
            });

            await batch.ExecuteAsync();

            return dataElement;
        }
```